Tests/NIOTransportServicesTests/NIOTSConnectionChannelTests.swift

```diff
@@ -487,4 +498,61 @@ class NIOTSConnectionChannelTests: XCTestCase {
             XCTFail("Unexpected error")
         }
     }
+
+    func testEarlyExitForWaitingChannel() throws {
+        let connectFuture = NIOTSConnectionBootstrap(group: self.group)
+            .channelOption(NIOTSChannelOptions.waitForActivity, value: false)
+            .connect(to: try SocketAddress(ipAddress: "127.0.0.1", port: 12345))
```

Comment:
I'm not super keen on trying this on port 12345. Should we use a temp directory with `AF_UNIX` so we can 'guarantee' that it won't work?

Reply:
Sure, why not.

Reply:
Please re-review.